src/System.Management.Automation/engine/runtime/Operations/ArrayOps.cs

```diff
@@ -12,6 +12,15 @@ namespace System.Management.Automation
 {
     internal static class ArrayOps
     {
+        internal static object AddObjectArray(object[] lhs, object rhs)
+        {
+            int newIdx = lhs.Length;
+            Array.Resize(ref lhs, newIdx + 1);
```

Comment:
If we look original issue the performance problem is only with _loops_. In the case it would be better to increase the size on an empirical value more than one - 4, 8, 16. This will significantly reduce the number of unnecessary array copies.

Reply:
This is called for every addition to the array, if we resize the array more than 1 then the output of `+=` will contain extra values. For example `$arr.Count` should be 1 but if we resize it with 4, 8, 16, etc then it will no longer be 1.

```powershell
$arr = @()
$arr += 1

$arr.Count
```

Reply:
Ah, it is not List :-)